### PR TITLE
refactor(explorer): extract dialogs

### PR DESCRIPTION
**Pull request by ingeridhellen**
## What does this pull request change?

The ContextMenu file was long and hard to get a grasp of, so I split it into multiple files. Also noticed that we were rendering 6 dialogs for each node in the tree, which seems a bit excessive, so I added an inline if around each render

### CODE REVIEW

File: packages/dm-core-plugins/src/explorer/components/context-menu/NodeRightClickMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        open={showMenu}
+        onClose={() => setShowMenu(false)}
+        anchorEl={anchorEl}
+        placement="bottom-start"
+        matchAnchorWidth={true}
+      >
+        {menuItems}
+      </Menu>
+
+      {dialogId === 'new-folder' && (
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/explorer/components/context-menu/NodeRightClickMenu.tsx:48]: This is old stuff, but maybe would be nice to use some enum or something for the dialog ids?

File: packages/dm-core-plugins/src/explorer/components/dialogs/DeleteDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      </Dialog.Header>
+      <Dialog.CustomContent>
+        Are you sure you want to delete the entity <b>{node.name}</b> of type{' '}
+        <b>{node.type}</b>?
+      </Dialog.CustomContent>
+      <Dialog.Actions>
+        <Button
+          color="danger"
+          onClick={async () => {
+            await DeleteAction(node, dmssAPI, setLoading)
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/explorer/components/dialogs/DeleteDialog.tsx:39]: This is old stuff, but does it make sense to have the action (DeleteAction) in a different file? It is only used inside this file, right?

**Review by eoaksnes — APPROVED** [review]
Very nice cleanup!

File: packages/dm-core-plugins/src/explorer/components/dialogs/DeleteDialog.tsx (hunk 2 above)
Comment (ingeridhellen) [reply packages/dm-core-plugins/src/explorer/components/dialogs/DeleteDialog.tsx:39]: Fix in new PR

File: packages/dm-core-plugins/src/explorer/components/context-menu/NodeRightClickMenu.tsx (hunk 1 above)
Comment (ingeridhellen) [reply packages/dm-core-plugins/src/explorer/components/context-menu/NodeRightClickMenu.tsx:48]: Will fix in new PR

**Pull request merged**